src/main/java/com/jcraft/jsch/OpenSshCertificateBuffer.java

```diff
@@ -0,0 +1,149 @@
+package com.jcraft.jsch;
+
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.jcraft.jsch.OpenSshCertificateUtil.isEmpty;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * A specialized buffer for parsing OpenSSH certificate data.
+ *
+ * <p>
+ * This class extends the base {@link Buffer} class to provide additional methods specific to
+ * parsing OpenSSH certificate format data structures, including string collections, key-value maps,
+ * and certificate-specific data types.
+ * </p>
+ *
+ * <p>
+ * The buffer follows the SSH wire format protocol for data serialization, where strings and byte
+ * arrays are prefixed with their length as a 32-bit integer.
+ * </p>
+ */
+class OpenSshCertificateBuffer extends Buffer {
+
+  private static final byte[] EMPTY_BYTE_ARRAY = {};
+
+
+  /**
+   * Creates a new OpenSSH certificate buffer from decoded certificate bytes.
+   *
+   * @param certificateByteDecoded the decoded certificate data
+   */
+  OpenSshCertificateBuffer(byte[] certificateByteDecoded) {
+    super(certificateByteDecoded);
+    s = 0;
+    index = certificateByteDecoded.length;
+
+  }
+
+  /**
+   * Reads a length-prefixed byte array from the buffer.
+   *
+   * @return the byte array data
+   */
+  public byte[] getBytes() {
+    int reqLen = getInt();
+    byte[] b = new byte[reqLen];
+    getByte(b);
+    return b;
+  }
+
+  /**
+   * Reads a string with the specified character encoding.
+   *
+   * @param charset the character encoding to use
+   * @return the decoded string
+   */
+  public String getString(Charset charset) {
+    return new String(getString(), charset);
+  }
```

Comment:
Can we drop this method and just use `Util.str2byte()` instead wherever you are using it?